### ISSUE TITLE
Spectre Import Failure

### ISSUE THREAD

**Issue by daprofessa19** [firefly-iii/firefly-iii#4615]
I am running Firefly III version 5.0.5, and my problem is:

Spectre import is failing with below message.

The import has suffered from an error it could not recover from. Apologies!

Job could not be started or crashed: The import routine crashed: Guzzle Exception: Client error: `GET https://www.saltedge.com/api/v4/logins/?from_id=0&customer_id=<REDACTED>` resulted in a `404 Not Found` response: {"error_class":"RouteNotFound","error_message":"Route 'api/v4/logins' was not found."}

Also, I am running my Firefly instance hosted on Yunohost. From within Yunohost, I can see than an upgrade is available as follows, 

Firefly III firefly-iii
from 5.0.5 to 5.4.6~ynh2

However, I have tried the upgrade and it is failing.

Please let me know what you think.

Thanks

**Comment by JC5** [comment]
The reason it doesn't work anymore is because v4 of the Spectre API has been deprecated and remove. You must upgrade Firefly III.

Since you're running Yunohost however, you'll have to ask them for support.

**Comment by daprofessa19** [comment]
Yes, I'm trying to upgrade this app. This functionality broke for me only in the last 1 or 2 weeks. I am trying to get support for this issue, and I noticed that you are a contributor to the Yunohost Firefly III repo.

It seems that the backup of the app is failing, and then the procedure does not continue beyond that. I have attached the errors that I am seeing.

2021-04-10 23:29:27,262: DEBUG - 1256 + ynh_print_log 'Source path '\''/etc/php/7.2/fpm/pool.d/firefly-iii.conf'\'' does not exist'
2021-04-10 23:29:27,263: DEBUG - 1256 + echo -e 'Source path '\''/etc/php/7.2/fpm/pool.d/firefly-iii.conf'\'' does not exist'
2021-04-10 23:29:27,263: DEBUG - 1256 + '[' 0 == 0 ']'
2021-04-10 23:29:27,263: DEBUG - 1256 + echo /etc/php/7.2/fpm/pool.d/firefly-iii.conf
2021-04-10 23:29:27,263: DEBUG - 1256 + grep --quiet /etc/fail2ban
2021-04-10 23:29:27,264: DEBUG - 1256 + return 1
2021-04-10 23:29:27,264: DEBUG - 1257 + ynh_exit_properly
2021-04-10 23:29:27,264: DEBUG - 1257 + local exit_code=1
2021-04-10 23:29:27,264: DEBUG - 1257 + '[' 1 -eq 0 ']'
2021-04-10 23:29:27,265: DEBUG - 1257 + trap '' EXIT
2021-04-10 23:29:27,265: DEBUG - 1257 + set +o errexit
2021-04-10 23:29:27,265: DEBUG - 1257 + set +o nounset
2021-04-10 23:29:27,265: DEBUG - 1258 + sleep 0.5
2021-04-10 23:29:27,767: DEBUG - 1759 + type -t ynh_clean_setup
2021-04-10 23:29:27,767: WARNING - 1781 Could not back up firefly-iii
2021-04-10 23:29:27,767: DEBUG - 1759 + exit 1
2021-04-10 23:29:27,767: WARNING - 1782 Nothing to save
2021-04-10 23:29:27,768: DEBUG - 1781 action [15434.1] executed in 1.675s
2021-04-10 23:29:27,868: DEBUG - + '[' 1 -eq 0 ']'
2021-04-10 23:29:27,868: WARNING - Backup failed, the upgrade process was aborted.
2021-04-10 23:29:27,869: DEBUG - + ynh_die '--message=Backup failed, the upgrade process was aborted.'
2021-04-10 23:29:27,871: DEBUG - + ret_code=1
2021-04-10 23:29:27,871: DEBUG - + echo 'Backup failed, the upgrade process was aborted.'
2021-04-10 23:29:27,872: DEBUG - + exit 1
2021-04-10 23:29:28,023: ERROR - Could not upgrade firefly-iii: An error occurred inside the app upgrade script
